lib/python/treadmill_aws/hostmanager.py

```diff
@@ -211,6 +211,11 @@ def delete_hosts(ec2_conn, ipa_client, hostnames):
                     record_value=hostname)
         except ipaclient.NotFoundError:
             pass
+        except ipaclient.ExecutionError:
+            # Invalid PTR record, delete bad data
+            ipa_client.force_delete_dns_record(
```

Comment:
so if there is force delete for all records that match the server, why bother with being nice? can we just use force always? cause the server is gone and will never come back..

Reply:
There can be a lot of data in an IPA DNS record. For example, you can add SRV records one-by-one and delete them singly, but it is also possible to modify all SRV records with a dnsrecord-mod call and delete all records with a dnsrecord-del --del-all call. I think that a more cautious approach will give us better habits with regard to these complex records. 

--del-all simply deletes all records, of all types, of that record name in the zone. This is safe to use with PTR records, because there are only PTR records in the reverse zones (x.x.x.in-addr.arpa.)